crates/wast/src/binary.rs

```diff
@@ -1108,7 +1228,159 @@ fn find_names<'a>(
+                Some(id.name())
+            }
+        }))
+    }
+
+    enum Name {
+        Type,
+        Func,
+        Module,
+        Component,
+        Instance,
+    }
+
+    let mut ret = ComponentNames::default();
+    ret.component = get_name(component_id, component_name);
+    for field in fields {
+        // Extract the kind/id/name from whatever kind of field this is...
+        let (kind, id, name) = match field {
+            ComponentField::Import(i) => (
+                match i.item.kind {
+                    ItemKind::Func(_) => Name::Func,
+                    ItemKind::Table(_) => panic!("table inside component"),
+                    ItemKind::Memory(_) => panic!("memory inside component"),
+                    ItemKind::Global(_) => panic!("global inside component"),
+                    ItemKind::Tag(_) => panic!("tag inside component"),
+                    ItemKind::Instance(_) => Name::Instance,
+                    ItemKind::Module(_) => Name::Module,
+                },
+                &i.item.id,
+                &i.item.name,
+            ),
+            ComponentField::NestedModule(m) => (Name::Component, &m.id, &m.name),
+            ComponentField::NestedComponent(c) => (Name::Component, &c.id, &c.name),
+            ComponentField::Instance(i) => (Name::Instance, &i.id, &i.name),
+            ComponentField::Type(t) => (Name::Type, &t.id, &t.name),
+            ComponentField::Func(f) => (Name::Func, &f.id, &f.name),
+            ComponentField::Alias(a) => (
+                match a.kind {
+                    ExportKind::Func => Name::Func,
+                    ExportKind::Table => panic!("table inside component"),
+                    ExportKind::Memory => panic!("memory inside component"),
+                    ExportKind::Global => panic!("global inside component"),
+                    ExportKind::Module => Name::Module,
+                    ExportKind::Instance => Name::Instance,
+                    ExportKind::Tag => panic!("tag inside component"),
+                    ExportKind::Type => Name::Type,
+                },
+                &a.id,
+                &a.name,
+            ),
+            ComponentField::Export(_) | ComponentField::Start(_) | ComponentField::Custom(_) => {
+                continue
+            }
+        };
+
+        // .. and using the kind we can figure out where to place this name
+        let (list, idx) = match kind {
+            Name::Func => (&mut ret.funcs, &mut ret.func_idx),
+            Name::Module => todo!("modules"),
+            Name::Component => (&mut ret.components, &mut ret.component_idx),
+            Name::Instance => (&mut ret.instances, &mut ret.instance_idx),
+            Name::Type => (&mut ret.types, &mut ret.type_idx),
+        };
+        if let Some(name) = get_name(id, name) {
+            list.push((*idx, name));
+        }
+
+        // Handle module locals separately from above
+        if let ComponentField::Func(f) = field {
```

Comment:
I don't think components can have functions with locals and such, so this may not be necessary?